doc/nrf/ug_matter_hw_requirements.rst

```diff
@@ -162,3 +166,299 @@ Values are provided in kilobytes (KB).
       +------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+-------------+
 
 ..
+
+.. _ug_matter_hw_requirements_layouts:
+
+Reference Matter memory layouts
+*******************************
+
+The following sections show how the conceptual :ref:`ug_matter_overview_architecture` translates to actual memory maps for each of the available :ref:`ug_matter_overview_architecture_integration_designs`.
+The memory values match `RAM and flash memory requirements`_ listed above.
+
+The following tabs show the memory maps for the development kits supported by the Matter protocol, including two memory maps for the :ref:`matter_weather_station_app`, which uses Nordic Thingy:53.
+
+.. tabs::
+
+   .. tab:: nRF52840 DK
+
+      The following memory map is valid for Matter applications running on the :ref:`nRF52840 DK <gs_programming_board_names>` (:ref:`nrf52840dk_nrf52840 <zephyr:nrf52840dk_nrf52840>`).
+
+      +--------------------------------+--------------------------+----------------+-------------------+---------------------+-----------------+-----------------+
+      | Memory region (incl. size)     | Partition                | Offset         | Size              | Partition elements  | Element offset  | Element size    |
```

Comment:
What about having both a "Partition name" (as in config) and a more user-friendly name e.g.:

"mcuboot" -> Bootloader (mcuboot) 
"mcuboot_primary" -> Application (mcuboot_primary)
"factory_data" -> Factory data (factory_data)
"settings_storage" -> Non-volatile storage (settings_storage)
"sram_primary" -> Static RAM (sram_primary)
"mcuboot_secondary" -> DFU slot (mcuboot_secondary) - I'm not certain about the name :/
"external_flash" -> Hmm.. maybe Free space (external_flash)

@Damian-Nordic any suggestions to last two names?

Reply:
- "Application DFU (mcuboot_secondary)"? 
- "Free space" sounds good. We could probably also just remove the partition. One can easily figure out based on offset + size of the partition and the size of the flash how much is left.